components/molecules/SuperlativeSelector/superlative-selector.tsx

```diff
@@ -50,23 +50,24 @@ const SuperativeSelector: React.FC<SuperlativeSelectorProps> = ({
     <div className="truncate" ref={ref}>
       <ContextFilterButton onClick={toggleFilter} isSelected={!!selected}>
         {selected ? (
-          <div className="flex" onClick={toggleFilter}>
-            <span className="inline-block text-dark-slate-10">Filtered by:</span>
-            <div className="block truncate pr-4 ml-1 text-light-slate-12">
-              {filterDescription}
-
-              <Icon
-                className="absolute right-1 top-1/2 transform -translate-y-1/2"
-                IconImage={cancelIcon}
-                onClick={() => {
-                  handleCancelClick();
-                  setIsOpen(false);
-                }}
-              />
+          <>
+            <div className="flex">
+              <span className="inline-block text-dark-slate-10">Filtered by:</span>
+              <button className="block truncate pr-4 ml-1 text-light-slate-12" onClick={toggleFilter}>
+                {filterDescription}
+                <Icon
+                  className="absolute right-1 top-1/2 transform -translate-y-1/2"
+                  IconImage={cancelIcon}
+                  onClick={() => {
+                    handleCancelClick();
+                    setIsOpen(false);
+                  }}
+                />
+              </button>
             </div>
-          </div>
+          </>
```

Comment:
Refactor (blocking): The React Fragment is unnecessary as the code block for when `selected` is true only has one React.Node.

```suggestion
            <div className="flex">
              <span className="inline-block text-dark-slate-10">Filtered by:</span>
              <button className="block truncate pr-4 ml-1 text-light-slate-12" onClick={toggleFilter}>
                {filterDescription}
                <Icon
                  className="absolute right-1 top-1/2 transform -translate-y-1/2"
                  IconImage={cancelIcon}
                  onClick={() => {
                    handleCancelClick();
                    setIsOpen(false);
                  }}
                />
              </button>
            </div>
```